packages/service-library/src/servicelib/logging_utils.py

```diff
@@ -456,6 +577,87 @@ def guess_message_log_level(message: str) -> LogLevelInt:
     return logging.INFO
 
 
-def set_parent_module_log_level(current_module: str, desired_log_level: int) -> None:
+def set_parent_module_log_level(
+    current_module: str, desired_log_level: LogLevelInt
+) -> None:
     parent_module = ".".join(current_module.split(".")[:-1])
     logging.getLogger(parent_module).setLevel(desired_log_level)
+
+
+def _store_logger_state(loggers: list[logging.Logger]) -> list[dict[str, Any]]:
+    """Store the original state of loggers for later restoration."""
+    return [
+        {
+            "logger": logger,
+            "handlers": logger.handlers.copy(),
+            "propagate": logger.propagate,
+        }
+        for logger in loggers
+    ]
+
+
+def _restore_logger_state(original_state: list[dict[str, Any]]) -> None:
+    """Restore loggers to their original state."""
+    for state in original_state:
+        logger = state["logger"]
+        logger.handlers.clear()
+        logger.handlers.extend(state["handlers"])
+        logger.propagate = state["propagate"]
+
+
+def _apply_comprehensive_logging_setup(
+    all_loggers: list[logging.Logger],
+    root_handler: logging.Handler,
+) -> list[dict[str, Any]]:
+    """
+    Apply comprehensive logging setup: clear all handlers, ensure propagation,
+    and set up root logger with the provided handler.
+
+    Returns the original logger state for restoration.
+    """
+    root_logger = logging.getLogger()
+
+    # Store original state for restoration
+    original_logger_state = _store_logger_state(all_loggers)
+
+    # Remove all handlers from all loggers and ensure propagation
+    loggers_modified = []
+    for logger in all_loggers:
+        if logger is root_logger:
+            continue
+
+        # Track what we're modifying for logging purposes
+        had_handlers = bool(logger.handlers)
+        had_propagate_disabled = not logger.propagate
+
+        if had_handlers or had_propagate_disabled:
+            loggers_modified.append(
+                {
+                    "name": logger.name,
+                    "had_handlers": had_handlers,
+                    "had_propagate_disabled": had_propagate_disabled,
+                    "handlers": [type(h).__name__ for h in logger.handlers],
+                }
+            )
+
+        # Clear handlers and ensure propagation
+        logger.handlers.clear()
+        logger.propagate = True
+
+    if loggers_modified:
+        _logger.info(
```

Comment:
this is still not affected by async ogs, right?

Reply:
right. I will move it down. good catch